src/beman/iterator_interface/CMakeLists.txt

```diff
@@ -1,55 +1,19 @@
-# src/beman/iterator_interface/CMakeLists.txt -*-CMake-*-
-#
+# cmake-format: off
+# src/beman/iterator_interface/CMakeLists.txt -*-cmake-*-
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# cmake-format: on
 
-add_library(beman.iterator_interface)
-add_library(beman::iterator_interface ALIAS beman.iterator_interface)
-
-target_sources(
-  beman.iterator_interface
-  PRIVATE
-  iterator_interface.cpp
-)
-
-include(GNUInstallDirs)
-include_directories(${CMAKE_CURRENT_SOURCE_DIR})
-
-target_include_directories(beman.iterator_interface PUBLIC
-  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
-  $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
-  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${CMAKE_LOWER_PROJECT_NAME}>  # <prefix>/include/scratch
-)
+# Ensure that iterator_interface gets compiled at least once.
```

Comment:
```suggestion
# Ensure that iterator_interface gets compiled at least once.
```

why at least once and not `exactly once`?

Reply:
Because we may have multiple cpp files in src/ on future. If each one includes the header, we compile the public interface multiple times. Today we especially add this cpp file to force building the public interface.

Reply:
We'll explore more for a convention in Beman libraries here - https://github.com/beman-project/optional26/pull/75

For now delivering a static library is good enough for next week's demo (even if it's actually header only).